core/src/toga/widgets/optioncontainer.py

```diff
@@ -183,7 +218,8 @@ def __init__(
             applied to the widget.
         :param content: The initial content to display in the OptionContainer. A list of
             2-tuples, each of which is the title for the option, and the content widget
-            to display for that title.
+            to display for that title; or 3-tuples, describing the title, content widget,
+            and icon for the content.
```

Comment:
Agreed it is inconsistent. My inclination for a consistency fix would be slightly different, though:
* Change the constructor to `(text, widget, icon, enabled)`
* Change `insert()` to be `(text, widget, *, icon, enabled)`

This is on the basis that icon is more likely to be used at time of construction than enabled (since it's in-practice required for use on mobile); neither argument was previously exposed at time of construction; and if the argument is being made keyword-only on `insert()`, there's no historical order interpretation to consider (although there's a backwards-incompatible change that will be caught by the change to being keyword-only).

The other approach that I contemplated was introducing a namedtuple that declares the actual arguments, so you could easily define a initial content of `content=[Option("foo", toga.Box(), enabled=False)]`. Not sure if that would be part of the global Toga namespace, (`toga.OptionContainer(content=[toga.Option("foo", toga.Box(), enabled=False)]`), or stored on the OptionContainer itself ((`toga.OptionContainer(content=[toga.OptionContainer.Option("foo", toga.Box(), enabled=False)]`).

Of course, it also raises the question of what is different between an `OptionItem` and this new namedtuple. I guess the ideal situation would be to allow the user to pass in `OptionItem` objects, but that will require a bit of surgery on the rest of the interface. It also raises the question of what to do with the `insert()` method - do we allow `insert(n, OptionItem(...))`?

This larger OptionItem change could be deferred to a later PR, though. Even if we do go down this path, we'd still need to continue to support the basic tuple on construction.

Reply:
> * Change the constructor to `(text, widget, icon, enabled)`
> * Change `insert()` to be `(text, widget, *, icon, enabled)`

The problem with that is what it does to the constructor documentation:

<img width="822" alt="Screenshot 2023-12-11 at 18 03 11" src="https://github.com/beeware/toga/assets/166954/0c9dab97-0417-4c17-9327-4f962e81cf81">

And if we add OptionItem as an alternative later, it'll get even worse.

So how about this:
* Give OptionItem a constructor with arguments `(text, widget, *, icon, enabled)`
* Make the OptionContainer constructor, `insert` and `append` all accept OptionItem. It looks like the alternative signatures of `insert` and `append` could be documented using `@overload`, though we could also consider deprecating the non-OptionItem versions.

This would also be extensible to any more attributes we add to OptionItem in the future.

Reply:
As with the Image constructor typing, this is one of those areas where my argument is that the problem is with the typing process, not with the actual interface. The actual interface isn't *that* complicated in practice. What is unwieldy is the syntax for formally describing that interface.

As I flagged in my last comment: using OptionItem will require a bit of surgery. The interface on `OptionItem` is the issue; it's not currently set up to be user-exposed. However, I'll take a look to see what can be done.